### DISCUSSION TITLE
Creating the fileless malware dataset that useful for deep learning model

### DISCUSSION THREAD

**Discussion by Aaronywd** [community/community#146423] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Now I currently collecting the dataset of the fileless malware based on the features PowerShell command, memory dump, network traffic and API call.  Previously, I have tried to setup the cuckoo sandbox for automatic analysis but failed. So, switch to the online sandbox, any.run, I get the report from the any.run. Then, how can I convert it into a dataset in csv based on the features suitable for a deep learning model?

**Comment by Opxl200X** [comment, marked as answer]
Hi @Aaronywd 👋

You're on the right track — creating a dataset for fileless malware based on dynamic features is a powerful approach for deep learning models. Since you're using Any.run, here’s a step-by-step guide to help you convert the sandbox reports into a CSV dataset:

---

### ✅ Workflow Overview

1. **Export reports** from Any.run in **JSON** format for easier parsing.
2. **Extract key features** such as:
   - PowerShell commands (count, suspicious keywords)
   - API calls (sequences or frequency of high-risk APIs)
   - Network traffic (DNS requests, IPs, domains)
   - Memory behavior (e.g., injection activity)

3. **Convert to CSV format** like this:

| Sample_ID | Num_PS_Commands | Suspicious_APIs | DNS_Requests | Memory_Inject | Label |
|-----------|------------------|------------------|----------------|----------------|--------|
| sample001 | 5                | 12               | 3              | 1              | 1      |

---

### 🧠 Sample Python Code

```python
import json, pandas as pd

data = []
for path in report_files:  # list of Any.run JSON reports
    with open(path) as f:
        r = json.load(f)
        data.append({
            "Sample_ID": r.get("task_id", "unknown"),
            "Num_PS_Commands": len(r.get("powershell", {}).get("commands", [])),
            "Suspicious_APIs": len(r.get("api", {}).get("calls", [])),
            "DNS_Requests": len(r.get("network", {}).get("dns", [])),
            "Memory_Inject": int("injection" in r.get("behavior", {}).get("summary", {})),
            "Label": 1  # Set manually or via rules
        })

pd.DataFrame(data).to_csv("fileless_dataset.csv", index=False)
